src/internal.c

```diff
@@ -7414,6 +7414,15 @@ static int wolfSSLReceive(WOLFSSL* ssl, byte* buf, word32 sz)
     if (recvd < 0) {
         switch (recvd) {
             case WOLFSSL_CBIO_ERR_GENERAL:        /* general/unknown error */
+                #if defined(OPENSSL_ALL) || defined(WOLFSSL_APACHE_HTTPD)
```

Comment:
Why add this to the `WOLFSSL_CBIO_ERR_GENERAL` case? Do these flags indicate read failure retry? Is WANT_READ the right way to handle that case? Maybe this be added to the `WOLFSSL_CBIO_ERR_WANT_READ` case?

Reply:
The changes are added to the `WOLFSSL_CBIO_ERR_GENERAL` case because that's the error (-1) that is returned from the callback. Here's the call Apache makes:
```
 476         /* If the read returns EAGAIN or success with an empty                  
 477          * brigade, return an error after setting the retry flag;               
 478          * SSL_read() will then return -1, and SSL_get_error() will             
 479          * indicate SSL_ERROR_WANT_READ. */                                     
 480         if (APR_STATUS_IS_EAGAIN(inctx->rc) || APR_STATUS_IS_EINTR(inctx->rc)   
 481                || (inctx->rc == APR_SUCCESS && APR_BRIGADE_EMPTY(inctx->bb))) { 
 482             BIO_set_retry_read(bio);                                            
 483             return -1;                                                          
 484         }
```
The flags set in `BIO_set_retry_read(bio)` indicate read failure retry. Initially I had made the changes on Apache's side to return `WOLFSSL_CBIO_ERR_WANT_READ`, though I figured it may be best to keep wolfSSL-specific changes minimal. 

Hope this clears it up a bit!